### PR TITLE
OCPBUGS-42483: PowerVS: update capi ibmcloud c6bcd313 for 4.17

### PR DESCRIPTION
**Pull request by hamzy**
Update the installer to use commit c6bcd313bce0fc9866e41bb9e3487d9f61c628a3 of cluster-api-provider-ibmcloud
This includes a couple of necessary Transit Gateway fixes.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -101,33 +101,33 @@ require (
...
+	k8s.io/apiextensions-apiserver v0.30.3
+	k8s.io/apimachinery v0.30.3
+	k8s.io/client-go v0.30.3
 	k8s.io/cloud-provider-vsphere v1.30.1
 	k8s.io/klog v1.0.0
 	k8s.io/klog/v2 v2.120.1
 	k8s.io/utils v0.0.0-20240310230437-4693a0247e57
 	libvirt.org/go/libvirtxml v1.10002.0
-	sigs.k8s.io/cluster-api v1.7.3
+	sigs.k8s.io/cluster-api v1.8.2
```
Comment (r4f4) [inline go.mod:123]: We should also bump the capi binary to the same version in `cluster-api/cluster-api`. Building against one version and running another is asking for trouble.
When we bump, we need to make sure all the other providers still work, so this PR needs to be thoroughly tested.

**Review by r4f4 — CHANGES_REQUESTED** [review]

File: go.mod (hunk 1 above)
Comment (hamzy) [reply go.mod:123]: @r4f4 Updated...

Comment (hamzy) [reply go.mod:123]: Weird it is here locally?
```
hamzy:hamzy-installer$ grep 'sigs.k8s.io/cluster-api v1.8.2' go.mod
	sigs.k8s.io/cluster-api v1.8.2
hamzy:hamzy-installer$ git status
On branch PowerVS-Update-417-CAPI-ibmcloud-c6bcd313
Your branch is up to date with 'origin/PowerVS-Update-417-CAPI-ibmcloud-c6bcd313'.
```

Comment (hamzy) [reply go.mod:123]: Forgot to also modify `cluster-api/cluster-api/go.mod` :(
Thanks!

**Comment by r4f4** [comment]
/hold

Let's fix the capi mismatch in the master branch first.

**Comment by r4f4** [comment]
Fix proposed in https://github.com/openshift/installer/pull/9055

**Comment by hamzy** [comment]
@r4f4 Can this progress now that #9055 has been approved?

**Comment by hamzy** [comment]
@gpei Hello, could you please label cherry-pick-approved?

**Pull request merged**